atomix/cluster/src/test/java/io/atomix/raft/RaftFailOverTest.java

```diff
@@ -118,19 +117,97 @@ public void shouldRecoverLeaderRestart() throws Throwable {
   }
 
   @Test
-  @Ignore("https://github.com/zeebe-io/zeebe/issues/4467")
-  public void testNodeCatchUpAfterCompaction() throws Exception {
+  public void shouldDoSnapshot() throws Exception {
     // given
-    raftRule.shutdownServer("1");
+    raftRule.appendEntries(128);
+
+    // when
+    raftRule.doSnapshot(100);
+
+    // then
+    assertThat(raftRule.allNodesHaveSnapshotWithIndex(100)).isTrue();
+  }
+
+  @Test
+  public void shouldCompactLogOnSnapshot() throws Exception {
```

Comment:
Isn't this testing the test setup rather than real implementation. Compaction is triggered by `RaftRule.RaftSnapshotListener`.

Reply:
I think you're totally right. But actually I would like to keep it, to make sure our test suite works also as expected. I could move it in a separate class called something like `RaftRuleTest` ? What do you think @deepthidevaki ?

Reply:
Do you really have to have the compaction logic in RaftRule? Isn't it used only in this test?

Reply:
Yes because with that I can test our bug we had, easily. Also other more complex scenarios

Reply:
Without this logic I wouldn't be able to test like shouldReplicateSnapshotOnJoin or shouldTakeMultipleSnapshotsAndReplicateSnapshotAfterRestart and also not what you mentioned below.

Reply:
Ok